sunpy/net/dataretriever/sources/noaa.py

```diff
@@ -52,6 +53,11 @@ def search(self, *args, **kwargs):
                 rowdict[key] = rowdict[key][0]
         rowdict['url'] = 'https://services.swpc.noaa.gov/json/solar-cycle/observed-solar-cycle-indices.json'
         rowdict['Instrument'] = 'NOAA-Indices'
+        # These results are not dependant on time, but we allow time as a
+        # parameter for easy searching, so remove time from the results table
+        # injected by GenericClient.
+        del rowdict['Start Time']
```

Comment:
this now raises a different error if no Time attrs is passed to this client. Really no time should in theory be passed as it does not depend on a given time. 

For example now with this"

```python
res = Fido.search(a.Instrument.noaa_indices)
~/sunpy_dev/sunpy-git/sunpy/net/dataretriever/sources/noaa.py in search(self, *args, **kwargs)
     57         # parameter for easy searching, so remove time from the results table
     58         # injected by GenericClient.
---> 59         del rowdict['Start Time']
     60         del rowdict['End Time']
     61         return QueryResponse([rowdict], client=self)

KeyError: 'Start Time'
```

before this PR  it did error too - but gave `NoMatchError: This query was not understood by any clients. Did you miss an OR?`

So shouldn't it be the case that no time should be passed? WIth any arbitrary time passed this works, just not when no time is given

Reply:
I have fixed this and added a time-less test.